src/NuGet.Core/NuGet.Resolver/ResolverUtility.cs

```diff
@@ -307,7 +307,7 @@ public static int GetLowestDistanceFromTarget(string packageId, HashSet<string>
                 level++;
 
                 // find the next level of dependencies
-                var dependencyIds = walkedPackages.SelectMany(package => package.Dependencies.Select(dependency => dependency.Id)).ToList();
+                var dependencyIds = walkedPackages.SelectMany(package => package.Dependencies.Select(dependency => dependency.Id));
```

Comment:
Same as the above. This makes a linear method now run with quadratic performance.